### PR TITLE
Refactor code to allow for Refinement

### LINKED ISSUE #173: Refactor CNS refinement modules to be executed after `topoaa`
With https://github.com/haddocking/haddock3/pull/152 we changed the data structure that `topoaa` exports which made it incompatible with the refinement modules.

These need to be refactored to be able to be executed after `topoaa`.

Additionally, the CNS code is expecting a single `input_pdb_filename_1`, meaning that the multiple molecules need to be merged as a single structure with multiple chains.

### PR DESCRIPTION
**Pull request by rvhonorato**
This is a re-do of #176.

I took a different, more complex approach making more derivative changes instead of changing it in the base (`topoaa`).

This PR is specifically for the changes to the code to allow for refinement, I'll add the sampling in a different issue/pr - https://github.com/haddocking/haddock3/issues/186

***
closes #173

### CODE REVIEW

File: src/haddock/libs/libcns.py [hunk 1]
```diff
@@ -22,10 +24,19 @@ def generate_default_header(protonation=None):
...
-    water_box = load_waterbox(cns_paths.water_box['boxtyp20'])
-
-    return (param, top, link, topology_protonation, trans_vec, tensor, scatter,
-            axis, water_box)
+    water_box = load_waterbox(cns_paths.water_box["boxtyp20"])
+
+    return (
+        param,
+        top,
+        link,
```
Comment (amjjbonvin) [inline src/haddock/libs/libcns.py:32]: Is there a water box generation step at the python level? Isn't that handled by the CNS module?

File: src/haddock/libs/libcns.py [hunk 2]
```diff
@@ -117,42 +126,42 @@ def write_eval_line(param, value, eval_line='eval (${}={})'):
...
 
 
 def load_ff_parameters(forcefield_parameters):
     """Add force-field specific parameters to its appropriate places."""
-    ff_param_header = f'{linesep}! FF parameters{linesep}'
-    ff_param_header += f'parameter{linesep}'
-    ff_param_header += f'  @@{forcefield_parameters}{linesep}'
-    ff_param_header += f'end{linesep}'
+    ff_param_header = f"{linesep}! FF parameters{linesep}"
+    ff_param_header += f"parameter{linesep}"
```
Comment (amjjbonvin) [inline src/haddock/libs/libcns.py:137]: Note that there is no need anymore to write the reading of `haddock.top` and `haddock.param` to the CNS input files. This is now handled by the `read_param.cns` script in each module

File: src/haddock/libs/libcns.py [hunk 3]
```diff
@@ -117,42 +126,42 @@ def write_eval_line(param, value, eval_line='eval (${}={})'):
...
 
 
 def load_ff_topology(forcefield_topology):
     """Add force-field specific topology to its appropriate places."""
-    ff_top_header = f'{linesep}! Toplogy{linesep}'
-    ff_top_header += f'topology{linesep}'
-    ff_top_header += f'  @@{forcefield_topology}{linesep}'
-    ff_top_header += f'end{linesep}'
+    ff_top_header = f"{linesep}! Toplogy{linesep}"
+    ff_top_header += f"topology{linesep}"
```
Comment (amjjbonvin) [inline src/haddock/libs/libcns.py:147]: Same comment as before - the only top file required to be passed to the CNS input files is for ligands if present.

**Review by amjjbonvin — APPROVED** [review]
Just test and working fine.

The `conformational_sampling` is not yet implemented, correct? At least I don't find it in the `default.cfg` file of the refinement module

File: src/haddock/libs/libcns.py (hunk 1 above)
Comment (rvhonorato) [reply src/haddock/libs/libcns.py:32]: It comes from this one, but I think this has always been there, is it needed?
```python
def load_waterbox(waterbox_param):
    """Add waterbox information."""
    water_header = f'{linesep}! Water box{linesep}'
    water_header += f'eval ($boxtyp20 = "{waterbox_param}" ){linesep}'

    return water_header
```

**Comment by rvhonorato** [comment]
> Just test and working fine.
> 
> The `conformational_sampling` is not yet implemented, correct? At least I don't find it in the `default.cfg` file of the refinement module

not yet! I'll do it in another PR after this one

File: src/haddock/libs/libcns.py (hunk 3 above)
Comment (rvhonorato) [reply src/haddock/libs/libcns.py]: removed!

File: src/haddock/libs/libcns.py (hunk 2 above)
Comment (rvhonorato) [reply src/haddock/libs/libcns.py]: removed!

**Review by joaomcteixeira — APPROVED** [review]

**Pull request merged**